speaker/main.go

```diff
@@ -411,8 +415,8 @@ func (c *controller) deleteBalancerProtocol(l log.Logger, protocol config.Proto,
 			return controllers.SyncStateSuccess
 		}
 	}
-	delete(c.svcIPs, name)
 	level.Info(l).Log("event", "serviceWithdrawn", "ip", c.svcIPs[name], "reason", reason, "msg", "withdrawing service announcement")
+	delete(c.svcIPs, name)
```

Comment:
nit: this should not be moved

Reply:
after the deletion c.svcIPs[name] will be nil?

Reply:
right, good catch!
Can you move this to a separate commit? I think it deserves being tracked, plus it wasn't clear looking at this changeset.